pennylane/qnodes/passthru.py

```diff
@@ -0,0 +1,145 @@
+
+        Like :class:`.BaseQNode._construct`, but does not use VariableRefs.
+
+        PassthruQNode does not have a Jacobian method, so it does not HAVE to use VariableRefs or scalar linear indexing of input parameters.
+        Two options:
+        1. Use VariableRefs anyway, re-use most BaseQNode methods.
+           Problem: after evaluating the VariableRefs, stacking sliced/indexed Tensors in Operation.parameters should somehow result in a Tensor, not an object array.
+        2. Do not use VariableRefs, call the qfunc each time :meth:`PassthruQNode.evaluate` is called (always mutable).
+           Problem: tensornet_tf requires variable_deps?
+
+
+        TODO rethink output_conversion? should require device to return things in a fixed form, but either as arrays or as AD Tensors, do conversion in interface (if necessary...)
+        """
+        # temporary queues for operations and observables
+        self.queue = []  #: list[Operation]: applied operations
+        self.obs_queue = []  #: list[Observable]: applied observables
+
+        # set up the context for Operator entry
+        if qml._current_context is None:
+            qml._current_context = self
+        else:
+            raise QuantumFunctionError(
+                "qml._current_context must not be modified outside this method."
+            )
+        try:
+            # turn off domain checking since PassthruQNode qfuncs can take any class as input
+            pennylane.operation.Operator.do_check_domain = False
+            # generate the program queue by executing the quantum circuit function
+            res = self.func(*args, **kwargs)
+        finally:
+            qml._current_context = None
+            pennylane.operation.Operator.do_check_domain = True
+
+        # check the validity of the circuit
+        self._check_circuit(res)
+        del self.queue
+        del self.obs_queue
+
+        # no VariableRefs, self.variable_deps is empty!
+        # generate the DAG
+        self.circuit = pennylane.circuit_graph.CircuitGraph(self.ops, self.variable_deps)
+
+        # check for operations that cannot affect the output
+        if self.properties.get("vis_check", False):
+            visible = self.circuit.ancestors(self.circuit.observables)
+            invisible = set(self.circuit.operations) - visible
+            if invisible:
+                raise QuantumFunctionError(
+                    "The operations {} cannot affect the output of the circuit.".format(invisible)
+                )
+
+    def evaluate(self, args, kwargs):
+        """Evaluate the quantum function on the specified device.
+
+        Args:
+            args (tuple[Any]): positional arguments to the quantum function (differentiable)
+            kwargs (dict[str, Any]): auxiliary arguments (not differentiable)
+
+        Returns:
+            float or array[float]: output measured value(s)
+        """
+        # exactly like BaseQNode.evaluate, but skips the output_conversion and
```

Comment:
same thought as above might apply here :thinking:

Reply:
I managed to remove the method by adding a bit more functionality in `BaseQNode.evaluate`.